invenio_records_resources/services/records/service.py

```diff
@@ -39,16 +41,30 @@ def indexer(self):
         """Factory for creating an indexer instance."""
         return self.config.indexer_cls(record_cls=self.config.record_cls)
 
-    @property
-    def search_engine(self):
+    # @property
+    # FIXME: tmp made a function due to lack of methods to pass the identity
+    # to the permission. Filter is calculated upon `search_cls` instantiation
+    # which happens in the engine constructor. Refactor should fix this too.
+    def search_engine(self, identity=None):
         """Factory for creating a search instance."""
         # This might grow over time
-        options = {
+        query_options = {
             "sorting": self.config.search_sort_options
         }
+
+        permission = self.permission_policy(
+            action_name="read", identity=identity
+        )
+
+        search_options = {
```

Comment:
minor: renaming `search_options` to `search_cls_options` ? query and search have such similar meaning...